traits/tests/test_trait_dict_object.py

```diff
@@ -121,6 +122,51 @@ def test_delitem_not_found(self):
             str(python_e.exception),
         )
 
+    if sys.version_info >= (3, 9):
```

Comment:
Would conditional skips of tests be better here?

Reply:
Possibly. I considered it, but right now we get a skip-free test run if all optional dependencies are installed, and I'd quite like to keep that property - it reassures me that all the tests have been run.